modules/generator/processor/servicegraphs/store/edge.go

```diff
@@ -27,6 +27,11 @@ type Edge struct {
 	// Additional dimension to add to the metrics
 	Dimensions map[string]string
 
+	// Additional labels for uninstrumented services
+	VirtualNode                           string // Indicates which service is the virtual node (client or server)
```

Comment:
instead of making these individual strings can we add them to the dimensions map above? and just allow the normal code to add them to the metrics?

Reply:
Definitely can be done. I've now added them as Dimensions if the feature is enabled.